### PR TITLE
Quest_parse rewrite

### PR DESCRIPTION
**Pull request by DantestyleXD**
[Probably a WiP]

For #388 

Outputs with this changes: https://pastebin.com/8AKqskGy

### CODE REVIEW

File: src/tools/quest_parser/main.cpp [hunk 1]
```diff
@@ -362,7 +405,7 @@ int main( int argc, char** argv )
   uint32_t i = 0;
   for( const auto& row : rows )
   {
-    Logger::info( "Generating {0}", row );
+    Logger::info( "Generating " + std::to_string( row ) );
```
Comment (NotAdam) [inline src/tools/quest_parser/main.cpp]: use format instead of constructing strings inline

File: src/tools/quest_parser/main.cpp [hunk 2]
```diff
@@ -407,7 +450,7 @@ int main( int argc, char** argv )
         "generated/" + questInfo->id + ".lua";
       if( system( command.c_str() ) == -1 )
       {
-        Logger::error( "Error executing java command:\n {0}\nerrno: {1}", command, std::strerror( errno ) );
+        Logger::error( "Error executing java command:\n" + command + "\nerrno: " + std::strerror( errno ) );
```
Comment (NotAdam) [inline src/tools/quest_parser/main.cpp]: use format instead of constructing strings inline

File: src/tools/quest_parser/main.cpp [hunk 3]
```diff
@@ -99,25 +112,25 @@ createScript( std::shared_ptr< Sapphire::Data::Quest >& pQuestData, std::set< st
...
         "( Entity::Player& player )\n"
         "   {\n"
-        "      player.eventPlay( this->getId(), " +
+        "      player.playScene( getId(), " +
         sceneId +
         ", 0,\n"
-        "         [&]( Entity::Player& player, uint32_t eventId, uint16_t param1, uint16_t param2, uint16_t param3 )\n"
-        "         {\n"
-        "         });\n"
+        "                       [&]( Entity::Player& player, const Event::SceneResult& result )\n"
```
Comment (NotAdam) [inline src/tools/quest_parser/main.cpp]: the indentation is off on these based on your example output

**Comment by DantestyleXD** [comment]
Actually output for playScene: https://pastebin.com/9un4pC1N

File: src/tools/quest_parser/main.cpp [hunk 4]
```diff
@@ -131,105 +144,133 @@ createScript( std::shared_ptr< Sapphire::Data::Quest >& pQuestData, std::set< st
...
+        cc++;
       }
     }
-    rewards += " };\n";
+    rewards += "  uint16_t RewardItemCount["+ std::to_string(cc) +"] = { "+ reward +" };\n";
+    reward.clear();
   }
 
-  if( !pQuestData->itemReward1.empty() )
+  if ( !pQuestData->itemReward1.at( 0 ) == 0 )
```
Comment (SapphireMordred) [inline src/tools/quest_parser/main.cpp]: style

File: src/tools/quest_parser/main.cpp [hunk 5]
```diff
@@ -131,105 +144,133 @@ createScript( std::shared_ptr< Sapphire::Data::Quest >& pQuestData, std::set< st
...
   {
-    rewards += "      static constexpr auto RewardItemOptional[] = { ";
-    for( size_t ca = 0; ca < pQuestData->itemReward1.size(); ca++ )
+    std::size_t cc = 0;
+    for ( size_t ca = 0; ca < pQuestData->itemReward1.size(); ca++ )
```
Comment (SapphireMordred) [inline src/tools/quest_parser/main.cpp]: and following. no space after control

File: src/tools/quest_parser/main.cpp [hunk 6]
```diff
@@ -285,34 +326,36 @@ createScript( std::shared_ptr< Sapphire::Data::Quest >& pQuestData, std::set< st
     scriptEntry += onEmoteStr;
   }
 
-  for( auto enemy : enemy_ids )
+  if(hasEnemies)
```
Comment (SapphireMordred) [inline src/tools/quest_parser/main.cpp]: missing spaces

**Comment by SapphireMordred** [comment]
Closing as this has been taken care of now.

**Pull request closed without merging**